packages/boot/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {statSync, readdirSync} from 'fs';
+import {join, resolve} from 'path';
+import {StringOrArr} from './types';
+const debug = require('debug')('@loopback/boot:utils');
+
+/**
+ * Takes an absolute directory path and returns all files / folders in it. Can optionally
+ * traverse nested folders.
+ *
+ * @param d Absolute path of directory to read
+ * @param nested Read nested directories in the given directory or not. Defaults to true.
+ */
+// tslint:disable-next-line:no-any
+export function readFolder(d: string, nested: boolean = true): any {
+  // If not reading nested directories, read the current one and return
+  if (!nested) {
+    try {
+      let files = readdirSync(d).map(f => join(d, f));
+      return files;
+    } catch (err) {
+      debug(`Skipping ${d} in if(!recursive) because error: ${err}`);
+      return [];
+    }
+  } else {
+    try {
+      // Recursively read nested directories
+      if (statSync(d).isDirectory()) {
+        return readdirSync(d).map(f => readFolder(join(d, f), nested));
+      } else {
+        return d;
+      }
+    } catch (err) {
+      debug(`Skipping ${d} in else because error: ${err}`);
+      return [];
+    }
+  }
+}
+
+/**
+ * Takes an array which may contain nested arrays and flattens them into a single array
+ *
+ * @param arr Nested array of elements to flatten
+ */
+// tslint:disable-next-line:no-any
+export function flatten(arr: any): any {
```

Comment:
Have you considered using a template argument here to avoid `any` and preserve type checks inside the function?

I think the following should work:

```ts
export function flatten<T>(arr: Array<Array<T> | T>): Array<T> {
  return arr.reduce(
    (flat, item) =>
      flat.concat(Array.isArray(item) ? flatten(item) : item),
    []);
}
```